rust-mode.el

```diff
@@ -1568,8 +1567,13 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
   (require 'rust-mode)
   (rust-mode))
 
-(defun rust--before-save-hook ()
-  (when rust-format-on-save (rust-format-buffer)))
+(defun rust--after-save-hook ()
+  (when rust-format-on-save
+    (condition-case rustfmt-err
+        (and
+         (rust-format-buffer)
+         (save-buffer))      
```

Comment:
Calling `save-buffer` from an `after-save-hook` seems fishy to me.  Maybe it would be better to just `message` or `warn` the failure and not propagate the exception.

Reply:
I think I remember my problem here. I don't know how to save a file without the message in the minibuffer.

Reply:
Using `after-save-hook` or `before-save-hook` ?